app/frontend/src/features/BugReport.tsx

```diff
@@ -0,0 +1,147 @@
+import { makeStyles } from "@material-ui/core";
+import { Error as GrpcError } from "grpc-web";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+
+import Alert from "../components/Alert";
+import Button from "../components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "../components/Dialog";
+import { BugIcon } from "../components/Icons";
+import TextField from "../components/TextField";
+import { service } from "../service";
+import { useAuthContext } from "./auth/AuthProvider";
+
+export interface BugReportFormData {
+  subject: string;
+  description: string;
+  steps: string;
+  results: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+  field: {
+    "& + &": {
+      marginBlockStart: theme.spacing(2),
+    },
+  },
+}));
+
+export default function BugReport() {
+  const classes = useStyles();
+  const [isOpen, setIsOpen] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    reset: resetForm,
+  } = useForm<BugReportFormData>();
+  const userId = useAuthContext().authState.userId;
+  const {
+    data: reportIdentifier,
+    error,
+    mutate: reportBug,
+    reset: resetMutation,
+  } = useMutation<string, GrpcError, BugReportFormData>((formData) =>
+    service.bugs.reportBug(formData, userId)
+  );
+
+  const handleClose = () => {
+    resetForm();
+    resetMutation();
+    setIsOpen(false);
+  };
+
+  const onSubmit = handleSubmit((data) => {
+    reportBug(data);
+  });
+
+  return (
+    <>
+      <Button onClick={() => setIsOpen(true)} startIcon={<BugIcon />}>
+        Report a bug
+      </Button>
+      <Dialog
+        aria-labelledby="bug-reporter"
+        open={isOpen}
+        onClose={handleClose}
+      >
+        <form onSubmit={onSubmit}>
+          <DialogTitle id="bug-reporter">Report a problem</DialogTitle>
+          <DialogContent>
+            {error && <Alert severity="error">{error.message}</Alert>}
+            {reportIdentifier && (
+              <Alert severity="success">
+                Thank you for reporting that bug and making Couchers better, a
+                report was sent to the devs! The bug ID is {reportIdentifier}
+              </Alert>
+            )}
+            <DialogContentText>
+              Please note that this information, as well as diagnostic
+              information including which page you are on, what browser you are
+              using, and your user ID will be saved to a public list of bugs.
+            </DialogContentText>
+            <TextField
+              id="bug-report-subject"
```

Comment:
A good thing I learnt while I was writing tests for this - we need to add an `id` for these text fields for material UI to link up text field labels properly so they remain accessible for screen readers. +1 on React Testing Library for encouraging these low hanging fruits to make sure I do it right!

Reply:
Should we open a low priority bug to add ids to all textFields then?  / Make it a required prop?

Reply:
Yeah that sounds like a good idea 👍